substrate/client/consensus/beefy/src/lib.rs

```diff
@@ -532,7 +586,7 @@ where
 	Err(ClientError::Backend(err_msg))
 }
 
-fn expect_validator_set<B, BE, R>(
+async fn expect_validator_set<B, BE, R>(
```

Comment:
I've been suggesting to use this function from the https://github.com/paritytech/polkadot-sdk/pull/1329, but looks like it isn't safe now, because it may block callers and if used on a critical path (e.g. during block import), we may halt the regular client operations. I suggest adding a comment to that function, mentioning that it uses `sleep` (asybc) inside, so shall be used with care

Reply:
done